### PR TITLE
oneDNN VTune profiling tutorial

### PR DESCRIPTION
**Pull request by louie-tsai**
## Description

This Jupyter Notebook demonstrates how to use VTune Profiler to profile oneDNN samples and find out performance bottlenecks.

## External Dependencies
pandas, matplotlib

## Checklist
Administrative
- [x] Review sample design with the appropriate [Domain Expert](https://github.com/oneapi-src/oneAPI-samples/wiki/Reviewers-and-Domain-Experts): Bogusevschi, Silviu, and other oneDNN engineers.
- [x] Sample has dependencies and/or includes Third-Party binaries. If yes inform the oneAPI Code Samples Project Manager: @JoeOster

Code Development
- [x] Implement coding guidelines and ensure code quality. [see wiki for details](https://github.com/oneapi-src/oneAPI-samples/wiki/General-Code-Guidelines)
- [x] Adhere to readme template 
- [x] Adhere to sample.json specification. https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-specification
- [x] Ensure/create CI test configurations for sample (ciTests field) https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-ci-test-object
- [ ] Run jsonlint on sample.json to verify json syntax. www.jsonlint.com

Security and Legal
- [x] OSPDT Approval (see @JoeOster for assistance)
- [x] Protex (see @JoeOster for assistance)
- [x] IPLDT Scan (see @JoeOster for assistance)

Review
- [ ] Review DPC++ code with Paul Peterseon. (GitHub User: pmpeter1)
- [ ] Review readme with Tom Lenth(@tomlenth) and/or Joe Oster(@JoeOster)
- [ ] Tested using Dev Cloud when applicable

### CODE REVIEW

**Comment by louie-tsai** [comment]
attached a pdf with results from previous run.
[tutorial_vtune_profiling_0512.pdf](https://github.com/oneapi-src/oneAPI-samples/files/6476065/tutorial_vtune_profiling_0512.pdf)

**Comment by JoeOster** [comment]
@louie-tsai PR description has been updated

**Comment by JoeOster** [comment]
Which domain expert did you use to review this with? Put it in the first line for the Admin checklist

**Comment by louie-tsai** [comment]
> Which domain expert did you use to review this with? Put it in the first line for the Admin checklist

I put name in the checklist, and I forwarded this MR to those oneDNN exports.

**Review by JoeOster — CHANGES_REQUESTED** [review]
Please take a look at yout TM and branding, oneDnn has specific rules

- The following names are allowed based on guidelines from Intel Legal (see namesdb.intel.com for more details)
    - oneDNN after oneAPI Deep Neural Network Library (oneDNN)

I sent you the TM&Branding report, please use that to look at anything that was flagged

File: Libraries/oneDNN/tutorials/README.md [hunk 1]
```diff
@@ -18,6 +18,7 @@ Third party program Licenses can be found here: [third-party-programs.txt](https
 | Component | [simple_model](tutorial_simple_model.ipynb)| A Jupyter notebook with step by step instructions on running oneDNN-based application on a GPU. |
 | Component | [verbose_jitdump](tutorial_verbose_jitdump.ipynb) | This Jupyter Notebook demonstrates how to use Verbose Mode and JIT Dump to profile oneDNN samples. |
 | Component | [analyze_isa_with_dispatcher_control](tutorial_analyze_isa_with_dispatcher_control.ipynb) | This Jupyter Notebook demonstrates how to use CPU Dispatch Control to generate JIT codes among different ISA on CPU and also analyze JIT kernels among ISAs.|
+| Component | [vtune profiling](tutorial_vtune_profiling.ipynb) | This Jupyter Notebook demonstrates how to use VTune Profiler to profile oneDNN samples and find out performance bottlenecks. |
```
Comment (JoeOster) [inline Libraries/oneDNN/tutorials/README.md:21]: vtune should have Intel in front of it on first use. 
Please add the "™" symbol after VTune

all instances of vtune, should be VTune unless used in a path

**Review by JoeOster — CHANGES_REQUESTED** [review]
Line#1 - Please add the "®" symbol after Intel
Line#3 - Please add the "®" symbol after Intel -
Note the first instance of Intel per heading style must include a "®" subsequent "Intel" do not
Line #3 does any of this apply?
The following names are allowed based on guidelines from Intel Legal (see namesdb.intel.com for more details)
Compute Library for Deep Neural Networks
Intel® Math Kernel Library for Deep Neural Networks
Intel Math Kernel Library for Deep Neural Networks after Intel® Math Kernel Library for Deep Neural Networks

Fixing these issues fulfill the IPLDT requirements

**Comment by louie-tsai** [comment]
@dmitryryintel
the memory bound number seems to be strange due to short profiling time.
Louie plans to make the samples run longer and then get the results again.

**Review by JoeOster — DISMISSED** [review]
approved for ci

**Comment by louie-tsai** [comment]
> @dmitryryintel
> the memory bound number seems to be strange due to short profiling time.
> Louie plans to make the samples run longer and then get the results again.

execute 20 times instead of 1 time to have more samples for profiling.
here is the result.
[tutorial_vtune_profiling - Jupyter Notebook.pdf](https://github.com/oneapi-src/oneAPI-samples/files/6519171/tutorial_vtune_profiling.-.Jupyter.Notebook.pdf)

**Comment by louie-tsai** [comment]
@JoeOster 

I couldn't reproduce the issue on my environment, and I got good result for "runipy tutorial_vtune_profiling.ipynb".
Can I get those output files like hotspot.csv from CI run?

regards
Louie

**Review by JoeOster — DISMISSED** [review]
approving for ci only

**Comment by louie-tsai** [comment]
@JoeOster 
Added error handlings for the ci testing issues.
please help me to re-test it.

**Review by JoeOster — APPROVED** [review]
approve for ci

**Comment by louie-tsai** [comment]
some system reqs for this VTune profiling.
/proc/sys/kernel/perf_event_paranoid needs to be set as 0.

**Comment by louie-tsai** [comment]
@JoeOster 
please help me to merge the PR if all look good to you.

thanks

**Pull request merged**